hedera-node/src/jmh/java/com/hedera/services/ledger/StakingActivityBench.java

```diff
@@ -0,0 +1,307 @@
+                final var nodeId = account.getStakedNodeAddressBookId();
+                summedStakesToReward.merge(
+                        EntityNum.fromLong(nodeId), roundedToHbar(account.totalStake()), Long::sum);
+            }
+        }
+        summedStakesToReward.forEach(
+                (num, amount) -> {
+                    final var info = app.stakingInfos().get().get(num);
+                    System.out.println(
+                            "- node"
+                                    + num.longValue()
+                                    + " has Merkle stakeToReward "
+                                    + info.getStakeToReward()
+                                    + " vs "
+                                    + amount
+                                    + " summed stakeToReward");
+                });
+    }
+
+    private long sumDetailRewards() {
+        var detailPendingRewards = new AtomicLong();
+        final var stakePeriodManager = app.periodManager();
+        final var curPeriod = stakePeriodManager.currentStakePeriod();
+        app.accounts()
+                .get()
+                .forEach(
+                        (num, account) -> {
+                            if (account.mayHavePendingReward()) {
+                                final var info =
+                                        app.stakingInfos()
+                                                .get()
+                                                .get(
+                                                        EntityNum.fromLong(
+                                                                account
+                                                                        .getStakedNodeAddressBookId()));
+                                final var detailReward =
+                                        rewardCalculator.computeRewardFromDetails(
+                                                account,
+                                                info,
+                                                curPeriod,
+                                                stakePeriodManager.effectivePeriod(
+                                                        account.getStakePeriodStart()));
+                                detailPendingRewards.getAndAdd(detailReward);
+                            }
+                        });
+        return detailPendingRewards.get();
+    }
+
+    @Benchmark
+    public void stakingActivities() {
+        // Also the staker in the case of a changed staking election
+        final var payerId = advanceToNextId();
+        // Also the stakee in the case of a staking election to account
+        final var countpartyId = advanceToNextId();
+        // Also the stakee in the case of a staking election to node
+        advanceI();
+        final var nodeI = Math.floorMod(i, numNodes);
+
+        rewardCalculator.reset();
+        sideEffects.reset();
+        ledger.begin();
+        // First we credit fees
```

Comment:
QQ: Are these needed here or just wanted their time included in benchmark ?

Reply:
They're necessary to get the interceptor and `TransactionalLedger` to work (otherwise for example, every `commit()`, the `0.0.800` account will be debited for *all rewards paid in the benchmark*---not just the current transaction).